server/datastore/mysql/software_titles_test.go

```diff
@@ -382,10 +384,85 @@ func testOrderSoftwareTitles(t *testing.T, ds *Datastore) {
 }
 
 func listSoftwareTitlesCheckCount(t *testing.T, ds *Datastore, expectedListCount int, expectedFullCount int, opts fleet.SoftwareTitleListOptions, returnSorted bool) []fleet.SoftwareTitle {
-	titles, count, _, err := ds.ListSoftwareTitles(context.Background(), opts)
+	titles, count, _, err := ds.ListSoftwareTitles(context.Background(), opts, fleet.TeamFilter{})
 	require.NoError(t, err)
 	require.Len(t, titles, expectedListCount)
 	require.NoError(t, err)
 	require.Equal(t, expectedFullCount, count)
 	return titles
 }
+
+func testTeamFilterSoftwareTitles(t *testing.T, ds *Datastore) {
+	ctx := context.Background()
+
+	team1, err := ds.NewTeam(ctx, &fleet.Team{Name: "team1"})
+	require.NoError(t, err)
+	team2, err := ds.NewTeam(ctx, &fleet.Team{Name: "team2"})
+	require.NoError(t, err)
+
+	user1, err := ds.NewUser(ctx, &fleet.User{Name: "user1", Password: []byte("test"), Email: "test1@email.com", GlobalRole: ptr.String(fleet.RoleAdmin)})
+	require.NoError(t, err)
+	user2, err := ds.NewUser(ctx, &fleet.User{Name: "user2", Password: []byte("test"), Email: "test2@email.com", Teams: []fleet.UserTeam{{Team: fleet.Team{ID: team1.ID}, Role: fleet.RoleAdmin}}})
+	require.NoError(t, err)
+	user3, err := ds.NewUser(ctx, &fleet.User{Name: "user3", Password: []byte("test"), Email: "test3@email.com", Teams: []fleet.UserTeam{{Team: fleet.Team{ID: team2.ID}, Role: fleet.RoleAdmin}}})
+	require.NoError(t, err)
+
```

Comment:
Can we add tests for the other user roles? Similar to how we're doing it for auth in the service layer? Something like the test cases in `server/service/software_test/TestServiceSoftwareInventoryAuth`.